docs/source/traitsui_user_manual/testing/discussions/debugging-gui-tests-at-runtime.rst

```diff
@@ -0,0 +1,15 @@
+Debugging GUI tests at runtime
+==============================
+
+To debug an issue with a GUI test, one can use the |UITester.delay| parameter
+to slow down the test, or they may use a Python debugger to inspect internal
+states while the test is being run. In both cases, developers should
+be aware that the GUI can react to any additional events that may be caused by
+these debugging activities, and this may cause the test to behave differently
+compared to running it in normal conditions.
+
+For example, if a test relies on a GUI widget having the keyboard input focus,
+a Python debugger may interfere with the test by stealing focus from the GUI
+when a break point occurs.
```

Comment:
Maybe add another example specific to when the delay is used? Something like "Additionally, if a test utilizes UITester.delay, the UI under test will respond to interactions by the user, e.g. a mouse click while the test is running.

This may be overkill though so feel free to ignore!

Reply:
I am leaning towards thinking this may be overkill at the moment. We can add more examples later though if it turns out the explanation was not clear enough (or maybe fix the explanation then).